nautobot/extras/tests/integration/test_customfields.py

```diff
@@ -196,7 +196,7 @@ def test_custom_field_advanced_ui(self):
         and checks it appears ONLY there!.
         """
         device = self.device
-        custom_field = CustomField(
+        custom_field = CustomField.objects.create(
```

Comment:
Not sure I'm a fan of this latest batch of changes - by calling `objects.create()` before we do a `save()` or `validated_save()` on the object, it gets written initially to the DB even if it later fails validation. Is this just needed because of removing `_fixup_empty_fields`?

Reply:
Yeah it is needed because we removed _fixup_empty_fields. 
```
cf = CustomField(type=CustomFieldTypeChoices.TYPE_TEXT, label="My Field", required=False)
cf.validated_save()
```
This sequence of code makes the validated_save() fail.

Reply:
*Why* though?? We have plenty of other models where you can omit a `slug` on create and it will automatically assign one, e.g.:

```python
>>> Location(name="hello", location_type=LocationType.objects.get(name="Root"), status=Status.objects.get(name="Active")).validated_save()
>>> Location.objects.get(name="hello").slug
'hello'
```

So what's missing here that it doesn't automatically work for CustomField?

Happy to help look into this tomorrow.

Reply:
Ok, I did a bit of digging, and it looks like the issue is that `AutoSlugField` gets its automatic value on `save()` (actually `pre_save()`), rather than on `clean()`. So if you don't set a `key` value before cleaning, you get:

```
django.core.exceptions.ValidationError: {'key': ['This field cannot be blank.', 'This key is not Python/GraphQL safe. Please do not start the key with a digit and do not use hyphens or whitespace']}
```

- The first error is because the field is defined as `blank=False`, where it should probably be changed to the `AutoSlugField` default of `blank=True`.
- The second error is because our new clean logic rejects a blank value as not matching the regex. This could probably be addressed by changing `check_if_key_is_graphql_safe` to accept blank values or changing our clean function to not even call that function if the key is blank.
    - There's a bit of trickiness here though as this would mean that a CF with `label="123 Main St"` and no specified key would, on save, get an auto-key of `123_main_st` which isn't GraphQL valid.
    - I *think* the approach here should be to augment `slugify_dashes_to_underscores` so that it automatically makes GraphQL valid slugs in such places, perhaps by just prepending `a` or `_` to any generated slug that would otherwise start with an invalid character?

Reply:
Thank you very much for looking into this! The latest commit has the following behavior:
```
>>> cf = CustomField(label="125 main st", required=False)
>>> cf.key
''
>>> cf.validated_save()
>>> cf.key
'a125_main_st'
```
I am prepending an "a" to the non graphql-safe `keys`. Since `slugify()` seems to remove the "_" I prepend to the `key`.